interfaces/builtin/avahi_control.go

```diff
@@ -135,7 +135,9 @@ func (iface *avahiControlInterface) AppArmorConnectedPlug(spec *apparmor.Specifi
 }
 
 func (iface *avahiControlInterface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
-	if !release.OnClassic {
+	// only apply slot snipped when running as application snap
+	// on classic, slot side can be system or application
+	if !isSlotSystemSlot(slot) {
```

Comment:
Same as above, maybe:
- implicitSystemSlot(slot) 
- ?
(also not great but…)

Reply:
Used proposal from Jamie, thank you :)